### PR TITLE
CB: Handle connection refused

### PR DESCRIPTION
**Pull request by drostron**
addresses: slamdata/slamdata#1159

### CODE REVIEW

File: couchbase/src/main/scala/quasar/physical/couchbase/fs/package.scala [hunk 1]
```diff
@@ -87,10 +95,15 @@ package object fs {
                      cm.info
                    ).as(
                      ().right
-                   ).handle { case ex: InvalidPasswordException =>
-                     invalidCredentials(
-                       "Unable to obtain a ClusterManager with provided credentials."
-                     ).right[NonEmptyList[String]].left
+                   ).handle {
+                     case ex: InvalidPasswordException =>
```
Comment (alissapajer) [inline couchbase/src/main/scala/quasar/physical/couchbase/fs/package.scala]: `ex` is not used

File: couchbase/src/main/scala/quasar/physical/couchbase/fs/package.scala [hunk 2]
```diff
@@ -50,6 +51,13 @@ package object fs {
     KeyValueStore[ResultHandle, Cursor, ?]
   )#M[A]
 
+  object CBConnectException {
+    def unapply(ex: Throwable): Option[Throwable] = ex.getCause match {
```
Comment (drostron) [inline couchbase/src/main/scala/quasar/physical/couchbase/fs/package.scala]: This would be better as `Option[ConnectException]d` — will update.

**Comment by alissapajer** [comment]
Do we have the test infrastructure to test something like this right now?

**Comment by drostron** [comment]
The necessary infrastructure exists. As far as I know testing a server startup scenario such as this would be new however.

**Comment by alissapajer** [comment]
@drostron was just checking on the build here, had a failure in `PrettifySpecs`. Added it to #1295 . I don't see how that could be related to this PR, but I'll let you restart that build to confirm.

**Comment by drostron** [comment]
Thanks for adding it to #1295 and agree that it's unrelated — _restarted_.

**Pull request merged**